crates/next-core/src/next_shared/transforms/next_strip_page_exports.rs

```diff
@@ -12,33 +12,37 @@ use turbopack_ecmascript::{CustomTransformer, EcmascriptInputTransform, Transfor
 use super::module_rule_match_js_no_url;
 
 /// Returns a rule which applies the Next.js page export stripping transform.
-pub async fn get_next_pages_transforms_rule(
+pub fn get_next_pages_transforms_rule(
     pages_dir: FileSystemPath,
     export_filter: ExportFilter,
     enable_mdx_rs: bool,
+    extra_conditions: impl IntoIterator<Item = RuleCondition>,
 ) -> Result<ModuleRule> {
     // Apply the Next SSG transform to all pages.
     let strip_transform =
         EcmascriptInputTransform::Plugin(ResolvedVc::cell(Box::new(NextJsStripPageExports {
             export_filter,
         }) as _));
-    Ok(ModuleRule::new(
+    let mut conditions = RuleCondition::all(vec![
         RuleCondition::all(vec![
-            RuleCondition::all(vec![
-                RuleCondition::ResourcePathInExactDirectory(pages_dir.clone()),
-                RuleCondition::not(RuleCondition::ResourcePathInExactDirectory(
-                    pages_dir.join("api")?,
-                )),
-                RuleCondition::not(RuleCondition::any(vec![
-                    // TODO(alexkirsz): Possibly ignore _app as well?
-                    RuleCondition::ResourcePathEquals(pages_dir.join("_document.js")?),
-                    RuleCondition::ResourcePathEquals(pages_dir.join("_document.jsx")?),
-                    RuleCondition::ResourcePathEquals(pages_dir.join("_document.ts")?),
-                    RuleCondition::ResourcePathEquals(pages_dir.join("_document.tsx")?),
-                ])),
-            ]),
-            module_rule_match_js_no_url(enable_mdx_rs),
+            RuleCondition::ResourcePathInExactDirectory(pages_dir.clone()),
+            RuleCondition::not(RuleCondition::ResourcePathInExactDirectory(
+                pages_dir.join("api")?,
+            )),
+            RuleCondition::not(RuleCondition::any(vec![
+                // TODO(alexkirsz): Possibly ignore _app as well?
+                RuleCondition::ResourcePathEquals(pages_dir.join("_document.js")?),
+                RuleCondition::ResourcePathEquals(pages_dir.join("_document.jsx")?),
+                RuleCondition::ResourcePathEquals(pages_dir.join("_document.ts")?),
+                RuleCondition::ResourcePathEquals(pages_dir.join("_document.tsx")?),
+            ])),
         ]),
+        module_rule_match_js_no_url(enable_mdx_rs),
+        RuleCondition::all(extra_conditions.into_iter().collect()),
+    ]);
+    conditions.flatten();
+    Ok(ModuleRule::new(
+        conditions,
         vec![ModuleRuleEffect::ExtendEcmascriptTransforms {
```

Comment:
Can we make this only use the `ReferenceType` condition? The rest is a hack really.

Reply:
we should be able to remove all of them.
(and only rely on whatever discovered all the `Endpoint`s, as opposed to this `./pages/*` condition matching).
`module_rule_match_js_no_url` might be needed if we have non-JS endpoints, but I think that isn't actually possible.